Eastern-Research-Group/mywaterway
# Feature/ts migration

## Related Issues:
* Continuing the process of converting the project to TypeScript

## Main Changes:
* Migrated the _LocationSearch_ context, along with several other contexts, and updated them to export a wrapper function that internalizes the call to `useContext`.
* Migrated the _WaterbodyInfo_ file.
* Migrated the _MapMouseEvents_ file.
* Migrated several other shared components.

## Steps To Test:
1. Test that the project compiles without errors.
2. The changes in the _LocationSearch_ file are likely the most widespread, and I changed some of the state variables to make them more consistent (see the `FetchState` type in that file). I took care to ensure the appropriate changes were made in the corresponding variables project-wide, but more testing will be helpful until those files get converted to TS as well.
